src/sig/sig.h

```diff
@@ -0,0 +1,138 @@
+/**
+ * \file sig.h
+ * \brief Header defining the API for generic OQS Signature
+ */
+
+#ifndef __OQS_SIG_H
+#define __OQS_SIG_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <oqs/rand.h>
+
+enum OQS_SIG_alg_name {
+	OQS_SIG_alg_default,
+};
+
+typedef struct OQS_SIG OQS_SIG;
+
+/**
+ * OQS signature object
+ */
+typedef struct OQS_SIG {
+
+	/**
+	 * PRNG
+	 */
+	OQS_RAND *rand;
+
+	/**
+	 * Specifies the name of the sig method
+	 */
+	char *method_name;
+
+	/**
+	 * Classical security in terms of the number of bits provided by the
+	 * signature method.
+	 */
+	uint16_t estimated_classical_security;
+
+	/**
+	 *  Equivalent quantum security in terms of the number of bits provided by the
+	 *  signature method.
+	 */
+	uint16_t estimated_quantum_security;
+
+	/**
+	 *  Private key length.
+	 */
+	uint16_t priv_key_len;
+
+	/**
+	 *  Public key length.
+	 */
+	uint16_t pub_key_len;
+
+	/**
+	 *  Maximum signature length.
+	 */
+	uint16_t max_sig_len;
+
+	/**
+	 * Named parameters for this key exchange method instance, if any.
+	 */
+	char *named_parameters;
+
+	/**
+	 * Opaque pointer for passing around instance-specific data
+	 */
+	void *params;
+
+	/**
+	 * Opaque pointer for passing around any computation context
+	 */
+	void *ctx;
+
+	/**
+	 * Pointer to a function for public and private signature key generation.
+	 *
+	 * @param s                Signature structure
+	 * @param priv             The private key
+	 * @param pub              The public key
+	 * @return                 1 on success, or 0 on failure
+	 */
+	int (*keygen)(const OQS_SIG *s, uint8_t **priv, uint8_t **pub);
```

Comment:
- Pass in OQS_RAND, not OQS_SIG.
- priv and pub should be uint8_t *, not uint8_t ** (i.e., up to the caller to provided memory, rather than callee to do a malloc)